src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -146,18 +146,31 @@ async function _is_object_version_fit(req, predicate, value) {
     return res;
 }
 
-async function has_bucket_policy_permission(policy, account, method, arn_path, req) {
+async function has_bucket_policy_permission(policy, account, method, arn_path, req, account_identifier_type = 'id') {
```

Comment:
Why did you choose to have id as default?
the id is currently only in NC.
Please add JSDoc to this function or a comment about the options of `account_identifier_type`. I think using a enum can help us.

Reply:
IIUC, the id is considered to solve issues regarding account name and thats the updated one. We are giving exception to `name` parameter.

> I think using a enum can help us
I feel it is for default argument, string is fine

Reply:
As you can see in the [Jira ticket](https://issues.redhat.com/browse/MCGI-281), it is not solving it completely.
I didn't understand the quote; was it not clear? I meant to add enum so it will be easier to understand the options that we can use.